main.py

```diff
@@ -46,6 +46,243 @@ def main():
+bot_commands = {}
+
+
+def helloworld(message):
+    current_chat = message['chat']['id']
+    data = {'chat_id': current_chat,
+            'text': 'Hello World!'}
+    tg.send_message(data)
+
+
+bot_commands["/helloworld"] = helloworld
+
+
+def source(message):
+    data = {'chat_id': current_chat,
+            'text': 'Inspect my insides! http://github.com/jarhill0/ABot'}
+    tg.send_message(data)
+
+
+bot_commands["/source"] = source
+
+
+def start(message):
+    user = message['from']['first_name']
+    bot_message = 'Hello %s! I am A Bot.' % user
+    bot_message += "\n\nHere's a list of my commands:" \
+                   "\n/redditposts [subreddit] - List 5 hot posts from /r/[subreddit]" \
+                   "\n/reddit [shortlink] - Get info on the linked Reddit post." \
+                   "\n/wa - Follow with a query to get information from WolframAlpha" \
+                   "\n/nextlaunch - Get information on the next SpaceX launch"
+    data = {'chat_id': current_chat,
+            'text': bot_message,}
+    tg.send_message(data)
+
+
+bot_commands["/start"] = start
+
+
+def help(message):
+    data = {'chat_id': current_chat,
+            'text': "Commands:\n"
+                    "\n/redditposts [subreddit] - List 5 hot posts from /r/[subreddit]"
+                    "\n/reddit [shortlink] - Get info on the linked Reddit post."
+                    "\n/wa - Follow with a query to get information from WolframAlpha"
+                    "\n/nextlaunch - Get information on the next SpaceX launch",}
+    tg.send_message(data)
+
+
+bot_commands["/help"] = help
+
+
+def settings(message):
+    data = {'chat_id': current_chat,
+            'text': "At the moment, I have no settings. Sorry."}
+    tg.send_message(data)
+
+
+bot_commands["/settings"] = settings
+
+
+def shrug(message):
+    data = {'chat_id': current_chat,
+            'text': '¯\_(ツ)_/¯',}
+    tg.send_message(data)
+
+
+bot_commands["/shrug"] = shrug
+
+
+def redditposts(message):
+    message_text = message.get('text', None)
+    command_block = message_text[message_text.index('/redditposts'):]
+    try:
+        subreddit = command_block.split(' ')[1]
+    except IndexError:
+        bot_message = 'Specify a subreddit after /redditposts (e.g. /redditposts funny)'
+    else:
+        bot_message, posts_dict = reddit.hot_posts(subreddit, 5)
+    finally:
+        data = {'chat_id': current_chat,
+                'text': bot_message,
+                'disable_web_page_preview': True}
+        tg.send_message(data)
+        try:
+            if posts_dict is not None:
+                reddit.add_posts_to_dict(current_chat, posts_dict)
+        except NameError:
+            pass
+
+
+bot_commands["/redditposts"] = redditposts
+
+
+def stop(message):
+    username = message['from'].get('username', None)
+    if (username == config.owner_un or username in config.owner_uns) and time.time() - \
+            message[
+                'date'] < 15:
+        sys.exit()
```

Comment:
bug in my code: this is all wrapped in a try-except, so I'm pretty sure the `sys.exit()` call fails every time.